### ISSUE TITLE
PDF tagging Pt.2

### ISSUE THREAD

**Issue by assembler55** [tecnickcom/tc-lib-pdf#174]
**Describe the bug**
The accessibility tagging is implemented and displayed in Adobe Acrobat, but the Paragraph tagging only tags the first line of text (I will post a picture in the comments below).
The Figure tag needs to contain the image/shape. In the provided examples, the shape is created and after that the Figure tag is set. That way, the Alt text does not describe the shape but the text element (picture in the comment section). 

**To Reproduce**
I'm working with example E063. 

**Expected behavior**
Whole paragraph needs to be tagged, not just the first line of text. 
The Figure tag is meaningless without image it belong to. The example tags the text within the shape, not the shape itself.

**Additional context**
The generalized PDF/UA example shows an error upon opening but can be viewed. 
The PDF/UA-1 example does not contain any tags.

**Additional notes**
Although not mentioned in the PDF/UA specification, the WCAG guidelines require the creation of Header bookmarks. Would it be possible to implement automatic bookmarking when using the H tags?

**Comment by assembler55** [comment]
The Reading order within Acrobat:
![image](https://github.com/user-attachments/assets/136139c7-1a8d-4bfc-9675-d43ad6614bdb)

The Structure tree within Acrobat: 
![image](https://github.com/user-attachments/assets/57c4dbe8-b430-4aed-b90c-ac036643cf9d)

The current implementation tags only the first line of text, not the whole paragraph.

**Comment by assembler55** [comment]
Hello, 

I have checked the edited example E063 and it looks perfect. 

The tag structure is well defined. 

One more note about PDF/UA standard. All not tagged content needs to be defined as Artefact, otherwise it won't conform to the PDF/UA standard. 

Header and footer should also be tagged as Artefacts (they should be ignored by screen-readers). 

The same applies to table lines and any other graphic without meaning (all decorative elements) - in short, nothing tagged by user. 

Would that be possible?

**Comment by nicolaasuni** [comment]
Since the original tagging issue is fixed I am going to close this issue.
Please feel free to open a more specific one, so it will be easier to track the progress.
